include/ros_tools/spline.h

```diff
@@ -45,6 +46,39 @@ namespace RosTools
 
         double findClosestSRecursively(const Eigen::Vector2d &point, double low, double high, int num_recursions) const;
     };
+
+    class Spline4D
```

Comment:
If Spline4D is in the same header, I think we should define the spline in the same source `cpp` file too. I'll make some changes